fendermint/testing/materializer/src/manifest.rs

```diff
@@ -0,0 +1,223 @@
+pub type NodeId = ResourceId;
+
+/// A human readable name for a relayer.
+pub type RelayerId = ResourceId;
+
+pub type SubnetMap = BTreeMap<SubnetId, Subnet>;
+pub type BalanceMap = BTreeMap<AccountId, Balance>;
+pub type CollateralMap = BTreeMap<AccountId, Collateral>;
+pub type NodeMap = BTreeMap<NodeId, Node>;
+pub type RelayerMap = BTreeMap<RelayerId, Relayer>;
+
+/// The manifest is a static description of a testnet.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Manifest {
+    /// All the accounts we want to act with across the entire subnet hierarchy.
+    ///
+    /// Each account will have its pair of private and public keys.
+    ///
+    /// In the rootnet, if we are dealing with Calibration, they will get their
+    /// initial balance from the Faucet, which should give 100 tFIL ("testnet" FIL),
+    /// which is why there is no definition for the account balances for the root.
+    ///
+    /// This would be different if we deployed a root in the test, e.g. using
+    /// Fendermint itself, in which case we could set whatever balance we wanted.
+    pub accounts: BTreeMap<AccountId, Account>,
+
+    /// Whether we use an existing L1 or create or own.
+    pub rootnet: Rootnet,
+
+    /// Subnets created on the rootnet.
+    pub subnets: SubnetMap,
+}
+
+/// Any potential attributes of an account.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Account {}
+
+/// Account balance.
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Balance(#[serde_as(as = "IsHumanReadable")] pub TokenAmount);
+
+/// Ways we can hook up with IPC contracts on the rootnet.
+///
+/// The rootnet is generally expected to be Calibration net,
+/// where IPC contracts are deployed from Hardhat, and multiple
+/// instances of the gateway exist, each with a different version
+/// and an address we learn after deployment.
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum IpcDeployment {
+    /// Use one of the existing deployments, given by the delegated address of
+    /// the Gateway and Registry contracts.
+    Existing {
+        #[serde_as(as = "IsHumanReadable")]
+        gateway: Address,
+        #[serde_as(as = "IsHumanReadable")]
+        registry: Address,
+    },
+    /// Deploy a new IPC contract stack using one of the accounts.
+    /// This can take a long time, but ensures we are testing with
+    /// contracts that have the same version as the client.
+    New { deployer: AccountId },
+}
+
+/// The rootnet, ie. the L1 chain, can already exist and be outside our control
+/// if we are deploying to Calibration, or it might be a chain we provision
+/// with CometBFT and Fendermint.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum Rootnet {
+    /// Existing L1 running outside our control.
+    ///
+    /// This implies using some sort of Faucet to get balances for the accounts.
```

Comment:
Can we record these requirements somewhere so we don't surprise ourselves at a later stage, when coming back to this?

Reply:
These are part of the testnet setup in the followup PR, it won't be a surprise.